web-app/src/pages/Dashboard/container.js

```diff
@@ -139,13 +145,74 @@ const DashboardContainer = Wrapped =>
       this.setState({ booking });
     };
 
-    requestHoliday = () => {
-      console.log('Holiday requested');
-      console.log('Starting', this.state.booking.start.format('Do MMMM YYYY'));
-      console.log('Finishing', this.state.booking.end.format('Do MMMM YYYY'));
-      console.log('Total', this.state.booking.duration, 'days');
+    submitHolidayRequest = () => {
+      const { booking, userDetails } = this.state;
+      const request = [];
+
+      for (let i = 0; i <= booking.end.diff(booking.start, 'days'); i++) {
+        const start = booking.start.clone();
+        request.push({
+          date: start.add(i, 'days').format('YYYY-MM-DD'),
+          dateCreated: moment().format('YYYY-MM-DD'),
+          halfDay: booking.isHalfday,
+          holidayId: 0,
+          holidayStatusDescription: 'Booked',
+          holidayStatusId: 1,
+          lastModified: moment().format('YYYY-MM-DD'),
+          employee: {
+            employeeId: userDetails.employeeId,
+            forename: userDetails.forename,
+            surname: userDetails.surname,
+            email: userDetails.email,
+            totalHolidays: 33,
+            startDate: [2014, 1, 1],
+            countryId: 1,
+            countryDescription: 'Northern Ireland',
+            employeeRoleId: 2,
+            employeeRoleDescription: 'System administrator',
+            employeeStatusId: 2,
+            statusDescription: 'Inactive',
+          },
+        });
+      }
+
+      requestHoliday(request).then(() => {
+        this.getTakenHolidays(userDetails.employeeId);
+        this.closeModal();
+      });
     };
 
+    updateHoliday = (cancel) => {
+      const { booking, userDetails } = this.state;
+      const request = {
+        date: booking.start.format('YYYY-MM-DD'),
+        dateCreated: moment().format('YYYY-MM-DD'),
+        halfDay: booking.isHalfday,
```

Comment:
The new changes to the server side will massively reduce this request, but works for now